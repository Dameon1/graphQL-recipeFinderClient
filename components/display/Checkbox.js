
import React from 'react';
import '../styles/checkbox.css';

export default class Checkbox extends React.Component {
  state = {
    isChecked: false,
  }

  toggleCheckboxChange = () => {
    const { handleCheckboxChange, label } = this.props;
    this.setState(({ isChecked }) => (
      { isChecked: !isChecked, }
    ));
    handleCheckboxChange(label);
  }

  render() {
    const { label } = this.props;
    const { isChecked } = this.state;

    return (
      <div className="checkbox" onClick={this.toggleCheckboxChange}>
        <label>
          <input
            type="checkbox"
            value={label}
            checked={isChecked}           
          />
          {label}
        </label>
      </div>
    );
  };
};
