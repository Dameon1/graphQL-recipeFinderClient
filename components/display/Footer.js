import React from 'react';
import '../styles/footer.css';

export default function Footer () {

  return (
    <div className='footer'>
      <p className="footerText">
        Powered by <a href="https://spoonacular.com/" className="footerLink" target="blank">@Spoonacular</a>
      </p>
    </div>
  )
};
