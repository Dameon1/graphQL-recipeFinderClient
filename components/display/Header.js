
import React from "react";
import '../styles/header.css';
import { signUserOut } from '../../actions/auth';
import { signingUserOut } from "../../actions/userActions";
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export class Header extends React.Component {

  handleSignOut() {
   this.props.dispatch(signingUserOut());
   this.props.dispatch(signUserOut());
  };

  render() {
    if (!this.props.loggedIn) {
      return (
        <div className='header'>
          <h1 className="headerText">
            what2eat
          </h1>
          <div className="navLinks">
            <Link to="/loginPage" >
              <button className="navLinkButtons">Login</button>
            </Link>
            <Link to="/signUp" >
              <button className="navLinkButtons">Sign up</button>
            </Link>  
            <Link to="/dashboard" >
              <button className="navLinkButtons" >Dashboard</button>
            </Link>
            <Link to="/" >
              <button className="navLinkButtons">Home</button>
            </Link>
          </div>
        </div>
    )}
      return (
        <div className='header'>
          <h1 className="headerText">
            what2eat
          </h1>
          <div className="navLinks">
            <Link to="/dashboard" >
              <button className="navLinkButtons">Dashboard</button>
            </Link>
            <Link to="/myRecipes" >
              <button className="navLinkButtons">My recipes</button>
            </Link> 
            <Link to="/" >
              <button className="navLinkButtons" onClick={ () => { this.handleSignOut() } }>Sign Out</button>
            </Link> 
          </div>
        </div>
    )
  };
};

const mapStateToProps = state => ({
  loggedIn: state.authReducer.loggedIn,
});

export default connect(mapStateToProps)(Header);
