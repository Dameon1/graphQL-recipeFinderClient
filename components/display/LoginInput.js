
import React from 'react';
import '../styles/loginInput.css';


export default class Input extends React.Component {

    componentDidUpdate(prevProps) {
        if (!prevProps.meta.active && this.props.meta.active) {
            this.input.focus();
        }
    };
   
    render() {
        let error;
        if (this.props.meta.touched && this.props.meta.error) {
            error = <div className = "form-error">{ this.props.meta.error }</div>;
        };

        let warning;
        if (this.props.meta.touched && this.props.meta.warning) {
            warning = (
                <div className = "form-warning">{ this.props.meta.warning }</div>
            );
        };

        return (
          <div className="form-input">
            { error }
            { warning }
            <input className="text-input"
                   id={ this.props.input.name }
                   placeholder={` ${this.props.input.name}`}
                   type={ this.props.type }
                   ref={ input => (this.input = input) }
            { ...this.props.input }
            />
          </div>
        );
    };
};
