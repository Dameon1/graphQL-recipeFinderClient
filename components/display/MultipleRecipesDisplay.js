
import React from 'react';
import "../styles/multipleRecipesDisplay.css";
import { fetchRecipesFromSpoonacularById } from '../../actions/spoonacularActions';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export function MultipleRecipesDisplay (props) {
 
  return (
    <div className=" recipeImageBox" 
         key={ props.index } 
         value={ props.id } >
      <h2 className="multipleRecipeTitle">{props.title}</h2>
      <img className="multipleRecipeImage" 
           src={ props.image } 
           alt={`Of ${props.title}`}>
      </img>
      <p>Ingredients used: { props.usedIngredientCount }</p>
      <p>Missing ingredients: { props.missedIngredientCount }</p>
      <Link to={ `/recipe/${props.id}` }>
        <button className="getRecipeButton" 
                onClick={ () => props.dispatch(fetchRecipesFromSpoonacularById(props.id)) }>
          Get Recipe
        </button>
      </Link> 
    </div>
    )
};
   
export default connect()(MultipleRecipesDisplay);
  