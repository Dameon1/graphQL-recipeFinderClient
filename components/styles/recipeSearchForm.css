
.recipesDisplayBox {
  display: flex;
  flex-wrap: wrap;
}

.recipeSearchForm {
  margin: 10px auto;
  width: 75%;
  border: solid #454545 2px;
  border-radius: 3px;
  border-top: 25px solid #454545;
  background-color:white;
}

.recipeSearchButton {
  display: block;
  margin: 10px auto;
  border-radius:5px;
  color: #454545;
  width:175px;
  background-color:#8FCFDE;
  height:35px;
}

.addIngredientForm {
  padding-bottom:10px;
  border: solid #454545 2px;
  border-radius: 2px;
  margin: 15px auto;
  width: 75%;
  text-align: center;
  border-top: 25px solid #454545;
  background-color: white;
} 
.addIngredientLabel {
  margin-top: 5px;
  display: inline-block;
  width: 100%;
  height : 25px;
}
.addIngredientInputField {
  width:60%;
  height : 25px;
  display:inline-block;
  margin:5px;
  background-color:rgb(241, 239, 239);
}
.addIngredientButton {
  font-size: 14px;
  text-align: center;
  display:inline-block;
  height : 25px;
  border-radius:5px;
  background-color:#8FCFDE;
  color:#454545
}

.addIngredientButton:hover {
  background-color:#454545;
  color:#8FCFDE;
}
.recipeSearchButton:hover {
  background-color:#454545;
  color:#8FCFDE;
}