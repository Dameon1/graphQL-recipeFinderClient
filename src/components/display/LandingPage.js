
import React from 'react';
import '../styles/landingPage.css';
import {Link} from 'react-router-dom';

export default function LandingPage() {

  return (
    <div className='landingPage'>
      <div className="landingPageBox">
        <h2 className='landingPageHeader'>You Have Ingredients?</h2>
        <p className='smallPrint'> (Find recipes for those ingredients!) </p>
        <p className="onBoarding">Sign up today to enjoy full features</p>
        <Link to='/signUp' >
          <button className="buttonStyle">SignUp</button>
        </Link>
        <p className="onBoarding">or preview the experience</p>
        <Link to='/dashboard' >
          <button className="buttonStyle">Preview</button>
        </Link>    
      </div>
    </div>
  )
};
