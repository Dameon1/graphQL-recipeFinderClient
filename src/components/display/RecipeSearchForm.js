
 import "../styles/recipeSearchForm.css";
 import React from 'react';
 import Checkbox from './Checkbox';
import { Query, renderToStringWithData } from 'react-apollo';

 import gql from 'graphql-tag';
 const Query_Recipes = gql`
  query($queryString: String!){
    fetchRecipesFromSpoonacular(queryString:$queryString){
      instructions
      usedIngredientCount
      image
      title
      id
    }
  }
`;

export default class RecipeSearchForm extends React.Component {
  constructor(props) {
    super(props);
    this.state= {
      items : ['Chicken','Eggs',"Cheese","Flour","Salt","Bacon","Honey","Butter","Cayenne","Sugar","Rice","Celery",
                'Agave','Cilantro','Cinnamon','Dijon','Lemon','Garlic'],
      queryString : "",
    }
  }

  componentWillMount = () => {
    this.selectedCheckboxes = new Set();
  }

  toggleCheckbox = label => {
    if (this.selectedCheckboxes.has(label)) {
      this.selectedCheckboxes.delete(label);
    } 
    this.selectedCheckboxes.add(label);
  }

  createCheckbox = label => {
    return (
      <Checkbox 
        label= { label }
        handleCheckboxChange= { this.toggleCheckbox }
        key= { label } />
    )
  }
  
  createCheckboxes = () => (
    this.state.items.map(this.createCheckbox)
  )

  addIngredents = (e) => {
    e.preventDefault();
    let value = this.input.value.trim();
    let recipeItemsToLowerCase = this.state.items.map(item => item.toLowerCase());
    let newIngredientToLowerCase = value.toLowerCase();
    if (recipeItemsToLowerCase.includes(newIngredientToLowerCase)) {
      return alert("Ingredient already included") 
    }
    if (value.length<3) { 
      return alert("Ingredient is invalid") 
    }
    this.setState({
      items:[...this.state.items,value],
    });
    this.input.value="";      
  }

  render() {
    return (
      <div>
        <form className="addIngredientForm" onSubmit = { (e)=>this.addIngredents(e) } > 
          <label htmlFor="addIngredient" 
                aria-labelledby="addIngredient"
                className="addIngredientLabel" >
            Add Your Special Ingredients Here
          </label>
          <input className="addIngredientInputField" 
              type="text" 
              placeholder=" Add Ingredient" 
              name="addIngredient" 
              id="addIngredient" 
              ref={ input => (this.input = input) }
          />
          <button className="addIngredientButton" type="submit" >Add</button>
        </form>
        <Query query={Query_Recipes}
               variables={ { queryString:this.state.queryString } }
        >
        {({ loading, error, data }) => { 
          {console.log(data)}     
          if (loading) return <p>Loading...</p>;
          if (error) return `${error.message}`;        
          return  (
            <div>Search Form Here</div>
          )
        }}
        </Query>
      </div>
    )
  }
}
// import { fetchRecipesFromSpoonacular } from '../../actions/spoonacularActions';
// import { userIsSearching } from "../../actions/userActions";
// import { connect } from 'react-redux';

//  export default class RecipeSearchForm extends React.Component {
//   constructor(props) {
//     super(props);
  
     
 
  

//   handleFormSubmit = formSubmitEvent => {
//     formSubmitEvent.preventDefault();
    
//     if(this.selectedCheckboxes.size === 0) { return alert('Please select some ingredients') }
//     for (const checkbox of this.selectedCheckboxes) {
//       this.setState({queryString: this.state.queryString + checkbox + ','});
//     };   
//     //let recipeString = this.state.queryString.slice(0,-1);
//     //this.props.dispatch(userIsSearching());
//     //this.props.dispatch(fetchRecipesFromSpoonacular(recipeString));
    
//   }
    
  

 
//   render() {
//     return (
//       <div>
//       <form className="addIngredientForm" onSubmit = { (e)=>this.addIngredents(e) } > 
//       <label htmlFor="addIngredient" 
//             aria-labelledby="addIngredient"
//             className="addIngredientLabel" >
//        Add Your Special Ingredients Here
//      </label>
//      <input className="addIngredientInputField" 
//             type="text" 
//             placeholder=" Add Ingredient" 
//             name="addIngredient" 
//             id="addIngredient" 
//             ref={ input => (this.input = input) }
//              />
//      <button className="addIngredientButton" type="submit" >Add</button>
//    </form>
      
    
                    
//               <form onSubmit = { this.handleFormSubmit } className = "recipeSearchForm" >
//                 {this.createCheckboxes()}
//                 <button className = "recipeSearchButton" type = "submit">Search</button>
//               </form> 
               
              
//           {/* {data.song.map((item, i) => (
//             <div key={i} className="songBox">
//               <img src={item.image} alt='fix'/>
//               <div>
//               <h3>{item.artist}</h3>
//               <a href={item.songUrl} target='blank'>{item.title}</a>
//               </div>
//             </div>
//           ))}          */}
//        </div> 
//       );
//     }}
//   </Query>
//        </div>
//        )
//   }
// }
 // {/* //       query={Query_Recipes}
  //       variables={this.state.queryString}
  //       //refetchQueries={[{ query: CURRENT_USER_QUERY }]}
  //     >
  //       {(query, { data, error, loading }) => { */}
 // {/* //        if(data){console.log(query)}
  //         return (
  //           <div>
  //                 <form className="addIngredientForm" onSubmit = { (e)=>this.addIngredents(e) } > 
  //         <label htmlFor="addIngredient" 
  //                aria-labelledby="addIngredient"
  //                className="addIngredientLabel" >
  //           Add Your Special Ingredients Here
  //         </label>
  //         <input className="addIngredientInputField" 
  //                type="text" 
  //                placeholder=" Add Ingredient" 
  //                name="addIngredient" 
  //                id="addIngredient" 
  //                ref={ input => (this.input = input) }
  //                 />
  //         <button className="addIngredientButton" type="submit" >Add</button>
  //       </form>
  //       <form onSubmit = { this.handleFormSubmit } className = "recipeSearchForm" >
  //         {this.createCheckboxes()}
  //         <button className = "recipeSearchButton" type = "submit">Search</button>
  //       </form> 
  //       </div> 
  //       )}}
  //     </Query>
  //   );
  // }
  // render() {
  //   return (
  //     <div>
  //       <form className="addIngredientForm" onSubmit = { (e)=>this.addIngredents(e) } > 
  //         <label htmlFor="addIngredient" 
  //                aria-labelledby="addIngredient"
  //                className="addIngredientLabel" >
  //           Add Your Special Ingredients Here
  //         </label>
  //         <input className="addIngredientInputField" 
  //                type="text" 
  //                placeholder=" Add Ingredient" 
  //                name="addIngredient" 
  //                id="addIngredient" 
  //                ref={ input => (this.input = input) }
  //                 />
  //         <button className="addIngredientButton" type="submit" >Add</button>
  //       </form>
  //       <form onSubmit = { this.handleFormSubmit } className = "recipeSearchForm" >
  //         {this.createCheckboxes()}
  //         <button className = "recipeSearchButton" type = "submit">Search</button>
  //       </form>         
  //     </div>
  //   );
  // };



// const mapStateToProps = state => ({
//    isSearching:state.recipeReducer.isSearching
// });

// export default connect(mapStateToProps)(RecipeSearchForm); *///}
