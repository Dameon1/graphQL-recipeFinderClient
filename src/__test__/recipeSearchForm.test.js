import React from "react";
import { shallow, mount } from "enzyme";
import { RecipeSearchForm } from "../components/display/recipeSearchForm";

describe("<RecipeSearchForm />", () => {
  it("renders without crashing", () => {
    shallow(<RecipeSearchForm />);
  });
});
