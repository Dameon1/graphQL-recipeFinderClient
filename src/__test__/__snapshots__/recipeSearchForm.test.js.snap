// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeSearchForm /> renders without crashing 1`] = `
<div>
  <styled.h2>
    Welcome to what2eat
  </styled.h2>
  <styled.form
    onSubmit={[Function]}
  >
    <styled.label
      aria-labelledby="addIngredient"
      htmlFor="addIngredient"
    >
      Add Your Special Ingredients Here
    </styled.label>
    <styled.input
      id="addIngredient"
      name="addIngredient"
      placeholder=" Add Ingredient"
      type="text"
    />
    <styled.button
      type="submit"
    >
      Add
    </styled.button>
  </styled.form>
  <styled.form
    onSubmit={[Function]}
  >
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Chicken"
      label="Chicken"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Eggs"
      label="Eggs"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Cheese"
      label="Cheese"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Flour"
      label="Flour"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Salt"
      label="Salt"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Bacon"
      label="Bacon"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Honey"
      label="Honey"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Butter"
      label="Butter"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Cayenne"
      label="Cayenne"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Sugar"
      label="Sugar"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Rice"
      label="Rice"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Celery"
      label="Celery"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Agave"
      label="Agave"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Cilantro"
      label="Cilantro"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Cinnamon"
      label="Cinnamon"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Dijon"
      label="Dijon"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Lemon"
      label="Lemon"
    />
    <Checkbox
      handleCheckboxChange={[Function]}
      key="Garlic"
      label="Garlic"
    />
    <styled.button
      type="submit"
    >
      Search
    </styled.button>
  </styled.form>
</div>
`;
